### DISCUSSION TITLE
Delay Action/Condition Nodes

### DISCUSSION THREAD

**Discussion by RedstoneParadox** [bitbrain/beehave#242] [Ideas]
A lot of times in games, enemies/npcs have actions that either have a cooldown or warmup time. These could be accomplished via new Delay Action and Condition nodes, which will each start a time when they are first ticked, return `SUCCESS` when they are ticked and the timer is done, and then restart the time the next time they are ticked. The difference between the Action and Condition nodes is that the Action returns `RUNNING` while the timer is still counting down while the condition returns `FAILURE` instead. The former would be used for actions that have a warmup and the latter for actions that have a cooldown.

**Comment by bitbrain** [comment]
I am wondering if this could be achieved via a decorator node - this then allows to wrap any node with a "DelayDecorator" and it does what you describe there.

**Reply by RedstoneParadox** [reply]
I can see that working for delaying execution, but I'm not sure how you'd have it fail if the timer wasn't done.

**Reply by RedstoneParadox** [reply]
To add to that, if you want to wait after completing a sequence of actions, it would be a bit weird to use a decorator.

**Comment by bitbrain** [comment]
Implemented through https://github.com/bitbrain/beehave/pull/276
